packages/@aws-cdk/aws-cloudwatch/lib/private/rendering.ts

```diff
@@ -55,6 +55,7 @@ function metricGraphJson(metric: IMetric, yAxis?: string, id?: string) {
 
     withExpression(expr) {
       options.expression = expr.expression;
+      if (expr.period && expr.period !== 300) { options.period = expr.period; }
```

Comment:
I omitted this originally because what would happen is the expression period would be propagated down to the queried metrics.

Have we confirmed this is really necessary? Does it actually solve the problem it says it's addressing?

Reply:
I mean... I know I suggested this fix in the linked bug report.

But I'm very much interested in WHY this change is necessary.

It might also be because of the `period` field in the enclosing graph, for example.

Looking for someone to do a slightly deeper dive and experimenting a bit with the combinations and telling us WHAT EXACTLY the behavior of CloudWatch is (and by extension, what the construct library needs to produce).

Reply:
Ok, I can create a minimal stack with a graph widget and deploy to my AWS account to see the difference between the two cases.

Reply:
Thanks!

Reply:
When you define a metric using `MathExpression`, CDK will generate a new invisible metric for each component of the expression. For example, let's say you want to create a widget with three metrics for a DynamoDB table: one for ConsumedReadCapacityUnits, one for ConsumedWriteCapacityUnits and one for the sum of the other two. The generated CloudFormation template will have five metrics altogether:

1. A visible one for ConsumedReadCapacityUnits
1. A visible one for ConsumedWriteCapacityUnits
1. An invisible one for ConsumedReadCapacityUnits
1. An invisible one for ConsumedWriteCapacityUnits
1. A visible one for ConsumedReadCapacityUnits + ConsumedWriteCapacityUnits

Metric 5 is the sum of metrics 3 and 4.

As it is now, if you specify a period in a `MathExpression`, that period is not being respected in the aggregate metric, but it is already being propagated down to its components. Note that this only affect the components of the math expression. Their visible counterparts remain unchanged. In the example above, only metrics 3 and 4 have the period changed. Since they are invisible, there is no noticeable effect in the final widget, which shows data at a default interval for the aggregate metric. For DynamoDB tables, for example, this default is 10 minutes.

This change solves the problem by overriding the period in the aggregate metric and not changing anything else in the final CloudFormation template. I also deployed stacks with different combinations of periods and overrides and the result is as expected.